lib/easypost/shipment.rb

```diff
@@ -13,6 +13,20 @@ def regenerate_rates(params = {})
     self
   end
 
+  # Create a Shipment.
+  def self.create(params = {}, api_key = nil)
+    shipment = params.reject { |k, _| [:carbon_offset].include?(k) }
```

Comment:
`params.slice(:carbon_offset)`

Reply:
@lanej this is definitely an idea. The other libraries (due to their language syntax) allow for an explicit param for this `with_carbon_offset` which is the path we chose there. We can either pull `carbon_offset` out of `param` which would deviate from all other library usage or add `with_carbon_offset` as another param at the end. Neither is ideal. We've decided to have it as a separate param for now to keep things consistent. Let us know if that's not the best path forward.

We do have plans to correct these kinds of items in the next major release - one of them being moving from positional arguments everywhere to keyword arguments everywhere so items are more explicit and we won't run into this issue in the future when expanding an interface. Let me know if you have thoughts.